plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/server/ServerSettingsWizardPageModel.java

```diff
@@ -472,7 +472,7 @@ public void setRoute(IRoute newRoute) {
 				this.deployProject, this.projects, 
 				this.sourcePath, this.podPath, this.useInferredPodPath,
 				getService(), getServiceItems(),
-				route, this.selectDefaultRoute, this.routesByProject,
+				newRoute, this.selectDefaultRoute, this.routesByProject,
```

Comment:
It would be nice if such cases could be found by the compiler. A warning like "The passesd value of parameter 'route' is always reaasigned before it is used" might help.
I did not add check ResourceUtils.areRelated(route, service) because list to select a route contains all project routes.